pygmt/clib/session.py

```diff
@@ -160,7 +160,7 @@ def session_pointer(self) -> ctp.c_void_p:
             If trying to access without a currently open GMT session (i.e., outside of
             the context manager).
         """
-        if not hasattr(self, "_session_pointer") or self._session_pointer is None:
+        if getattr(self, "_session_pointer", None) is None:
```

Comment:
`if not hasattr(self, "_session_pointer") or self._session_pointer is None:` is equivalent to `if getattr(self, "_session_pointer", None) is None:`.